### PR TITLE
DROID-2219 Sentry | Release 6

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: app/src/main/java/com/anytypeio/anytype/ui/editor/EditorFragment.kt [hunk 1]
```diff
@@ -1309,9 +1309,10 @@ open class EditorFragment : NavigationFragment<FragmentEditorBinding>(R.layout.f
     }
 
 
-    private fun proceedWithScrollingToActionMenu(command: Command.ScrollToActionMenu): Unit {
+    private fun proceedWithScrollingToActionMenu(command: Command.ScrollToActionMenu) {
```
Comment (konstantiniiv) [inline app/src/main/java/com/anytypeio/anytype/ui/editor/EditorFragment.kt:1312]: https://sentry.anytype.io/organizations/anytype/issues/50734/?project=5

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt [hunk 2]
```diff
@@ -1017,15 +1017,16 @@ class HomeScreenViewModel(
     private fun proceedWithChangingType(widget: Id) {
         Timber.d("onChangeWidgetSourceClicked, widget:[$widget]")
         val curr = widgets.value.orEmpty().find { it.id == widget }
-        if (curr != null) {
+        val sourceId = curr?.source?.id
+        if (curr != null && sourceId != null) {
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:1021]: https://sentry.anytype.io/organizations/anytype/issues/50734/?project=5

File: app/src/main/java/com/anytypeio/anytype/ui/types/create/CreateObjectTypeFragment.kt [hunk 3]
```diff
@@ -73,9 +74,13 @@ class CreateObjectTypeFragment : BaseBottomSheetComposeFragment() {
                     findNavController().popBackStack()
                 }
                 is CreateObjectTypeViewModel.Navigation.SelectEmoji -> {
-                    findNavController().navigate(
-                        R.id.openEmojiPicker, bundleOf(ARG_SHOW_REMOVE_BUTTON to it.showRemove)
-                    )
+                    runCatching {
```
Comment (konstantiniiv) [inline app/src/main/java/com/anytypeio/anytype/ui/types/create/CreateObjectTypeFragment.kt:77]: https://sentry.anytype.io/organizations/anytype/issues/50627/?project=5&referrer=slack

File: app/src/main/java/com/anytypeio/anytype/ui/types/edit/TypeEditFragment.kt [hunk 4]
```diff
@@ -79,7 +80,11 @@ class TypeEditFragment : BaseBottomSheetComposeFragment() {
                     findNavController().popBackStack()
                 }
                 is TypeEditViewModel.Navigation.SelectEmoji -> {
-                    findNavController().navigate(R.id.openEmojiPicker)
+                    runCatching {
```
Comment (konstantiniiv) [inline app/src/main/java/com/anytypeio/anytype/ui/types/edit/TypeEditFragment.kt:83]: https://sentry.anytype.io/organizations/anytype/issues/50627/?project=5&referrer=slack

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sharing/AddToAnytypeViewModel.kt [hunk 5]
```diff
@@ -237,7 +237,12 @@ class AddToAnytypeViewModel(
 
             if (targetSpaceView != null && targetSpaceId != null) {
                 val paths = uris.mapNotNull { uri ->
-                    fileSharer.getPath(uri)
+                    try {
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sharing/AddToAnytypeViewModel.kt:240]: https://sentry.anytype.io/organizations/anytype/issues/50624/?project=5&referrer=slack

File: presentation/src/main/java/com/anytypeio/anytype/presentation/editor/EditorViewModel.kt [hunk 6]
```diff
@@ -6038,7 +6038,13 @@ class EditorViewModel(
 
         controlPanelInteractor.onEvent(ControlPanelMachine.Event.Mentions.OnMentionClicked)
 
-        val target = blocks.first { it.id == focus.value }
+        val target = blocks.find { it.id == focus.value }
```
Comment (konstantiniiv) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/editor/EditorViewModel.kt:6041]: https://sentry.anytype.io/organizations/anytype/issues/50505/?project=5&referrer=slack

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/common/Markup.kt [hunk 7]
```diff
@@ -139,8 +139,8 @@ fun Markup.toSpannable(
     }
 }
 
-private fun isRangeValid(mark: Markup.Mark): Boolean {
-    return mark.from >= 0 && mark.to >= 0 && mark.from < mark.to
+private fun isRangeValid(mark: Markup.Mark, textLength: Int): Boolean {
```
Comment (konstantiniiv) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/common/Markup.kt:142]: https://sentry.anytype.io/organizations/anytype/issues/50209/?project=5&referrer=slack

**Review by uburoiubu — APPROVED** [review]

**Pull request merged**